### PR TITLE
feat(composer): auto pin project requires

### LINKED ISSUE #2678: Composer support auto rangeStrategy for apps
https://getcomposer.org/doc/04-schema.md#type

### PR DESCRIPTION
**Pull request by rarkins**
Pins `require` packages if type=project and rangeStrategy=auto.

Closes #2678

### CODE REVIEW

**Comment by rarkins** [comment]
@evilebottnawi please check it for me

File: lib/manager/composer/range.js [hunk 1]
```diff
@@ -20,6 +26,11 @@ function getRangeStrategy(config) {
     logger.trace({ dependency: depName }, 'Pinning require-dev');
     return 'pin';
   }
+  if (composerJsonType === 'project' && depType === 'require') {
```
Comment (evilebottnawi) [inline lib/manager/composer/range.js]: In `PHP` there are a lot of other types (`wordpress-plugin`, `wordpress-theme` `metadata`, `composer-plugin` and etc). If project doesn't set `type` it is library, if set it is not library. 

I think better use this check:
```js
res.composerJsonType = packageJson.type || 'library';
```

```js
if (composerJsonType !== 'library' && depType === 'require') {
}
```

Example `wordpress-theme` use `wordpress-theme` value for `type` and `wordpress-theme` closer to `app` than library.

Comment (rarkins) [reply lib/manager/composer/range.js]: I was a little worried about assuming every non-library type was an app though. Are `metapackage` and `composer-plugin` types both considered to be apps?

Comment (evilebottnawi) [reply lib/manager/composer/range.js]: @rarkins hm, based on my knowledge `metapackage` and `composer-plugin` closest to `library` type, but they very rare used in real world, i think we can consider `metapackage` and `composer-plugin` as `library` too, i.e. `const isLibrary = ["library", "metapackage", "composer-plugin"].includes(res.composerJsonType)`

**Pull request merged**